### DISCUSSION TITLE
Geting routes with .yml

### DISCUSSION THREAD

**Discussion by Stanisla** [onthegomap/planetiler#1085] [Q&A]
Hi.
I'm trying to get routes as [https://wiki.openstreetmap.org/wiki/Key:route](url)
This is the custom schema i'm trying:
> ```
> schema_name: Monaco Router Schema
> schema_description: Attempt for routes
> attribution: <a href="https://www.openstreetmap.org/copyright" target="_blank">&copy;
>   OpenStreetMap contributors</a>
> args:
>   force: true   #always overwrite output 
>   output: ../monaco.pmtiles # where to output tiles,name and type
>   download: true # false (download sources)
> sources:
>   osm:
>     type: osm
>     url: geofabrik:monaco
> layers:
> - id: routes
>   features:
>   - source: osm
>     geometry: polygon
>     include_when:
>       route: __any__
>     attributes:
>       - key: route
> - id: routesLine
>   features:
>   - source: osm
>     geometry: line
>     include_when:
>       route: __any__
>     attributes:
>       - key: route
> ```

It's only returning osm way like [https://www.openstreetmap.org/way/166399542](url) but i can't get relations, like [https://www.openstreetmap.org/relation/2207095#map=15/43.74036/7.42794](url)
The last is what i`m really interested.
Any idea what i'm doing wrong?

**Comment by msbarry** [comment]
Yaml schemas don't support relations yet, see #351 and #857 for 2 possible paths forward

**Comment by Stanisla** [comment]
Thanks, i was going crazy ;)
From what I've read, we have to wait
On the other hand, I can't use:
> ```
> geometry: any
> ```
What is the correct way?

**Reply by msbarry** [reply]
you should be able to use `geometry: any` or omit geometry since `any` is the default with a snapshot build (it was added after the last release). That just makes it so that any point, line, or polygon from the input is passed through without filtering on a specific geometry type.

**Reply by Stanisla** [reply]
With  `geometry: any`  throws this error:

>'Exception in thread "main" java.lang.IllegalArgumentException: Cannot deserialize value of type `com.onthegomap.planetiler.custommap.configschema.FeatureGeometry` from String "any": not one of the values accepted for Enum class: [centroid, polygon_point_on_surface, line, polygon_centroid, polygon, line_centroid, polygon_centroid_if_convex, point, point_on_line]

If geometry is omitted,this is ther error:
>Exception in thread "main" java.lang.IllegalArgumentException: Missing required creator property 'geometry' (index 4)

**Reply by msbarry** [reply]
What version of planetiler are you running?

**Comment by Stanisla** [comment]
This one
>'0:00:00 INF - Planetiler build git hash: a74f274c3162cd547c9b19a9a5e8705982a61755
0:00:00 INF - Planetiler build version: 0.8.2
0:00:00 INF - Planetiler build timestamp: 2024-07-02T11:38:28.480Z
